source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -670,25 +613,16 @@ bool CIrrDeviceSDL::run()
 				}
 #endif
 				irrevent.EventType = irr::EET_KEY_INPUT_EVENT;
-
-				if (isNoUnicodeKey(key))
-					irrevent.KeyInput.Char = 0;
-				else
-					irrevent.KeyInput.Char = SDL_event.key.keysym.sym;
-
 				irrevent.KeyInput.Key = key;
 				irrevent.KeyInput.PressedDown = (SDL_event.type == SDL_KEYDOWN);
 				irrevent.KeyInput.Shift = (SDL_event.key.keysym.mod & KMOD_SHIFT) != 0;
 				irrevent.KeyInput.Control = (SDL_event.key.keysym.mod & KMOD_CTRL ) != 0;
-
-				// SDL2 no longer provides unicode character in the key event so we ensure the character is in the correct case
-				// TODO: Unicode input doesn't work like this, should probably use SDL_TextInputEvent
-				bool convertCharToUppercase =
-					irrevent.KeyInput.Shift != !!(SDL_event.key.keysym.mod & KMOD_CAPS);
-
-				if (convertCharToUppercase)
-					irrevent.KeyInput.Char = irr::core::locale_upper(irrevent.KeyInput.Char);
-
+				// These keys are handled differently in CGUIEditBox.cpp (may become out of date!)
+				// Control key is used in special character combinations, so keep that too
+				// Pass through the keysym only then so no extra text gets input
+				irrevent.KeyInput.Char = 0;

```

Comment:
This completely wipes out the case-sensitive ascii character (Key is case-insensitve, e.g. KEY_KEY_T) which means it's no longer possible to type text in the chat/command window. What's the intent here?

Reply:
Note: The existing code is (afaict) also very broken and many ascii characters fail to work in chat (because the KeyMap is wrong). I have a patch that fixes it for the minetest web version, but only supports ascii.

Reply:
Apologies, I just realized SDL on emscripten may be broken with regard to the SDL_TEXTINPUT event, which is the real issue here. I'll investigate further.